### PR TITLE
Add a warning when list initialization involves mismatched parSafe settings

### PR DESCRIPTION
**Pull request by lydia-duncan**
This tripped up a user this week, so we figured a warning would be useful.

Enables the warning to be controlled by a flag, `warnForListParsafeMismatch`, in case users are intentionally relying on this behavior.

Removed some incorrect documentation about the behavior of the copy initializer, and updated the code within to reflect what the compiler actually does with the type (even though the code itself doesn't impact that, in case adjustments to dyno rely on it in the future).

While here, also fixed a typo I noticed in the existing init= test (which thankfully doesn't change its behavior)

Adds a new test locking in the warning based on the user's reproducer and updates the existing init= test to include the warning when the test exercises it.  Also updated a list error message test to not worry about line number changes in the List module.

Passed a full paratest with futures

### CODE REVIEW

**Review by mppf — APPROVED** [review]

**Comment by astatide** [comment]
This definitely would have helped me avoid tripping up :joy: I think that's a great warning!

File: modules/standard/List.chpl [hunk 1]
```diff
@@ -423,11 +430,15 @@ module List {
...
-      // set parSafe to false if it was not already provided in lhs type
+      // set parSafe to other.parSafe if it was not already provided in lhs type
       this.parSafe = if this.type.parSafe != ?
                      then this.type.parSafe
-                     else false;
+                     else other.parSafe;
 
+      if (this.parSafe != other.parSafe && warnForListParsafeMismatch) {
+        compilerWarning("initializing between two lists with different " +
+                        "parSafe settings");
```
Comment (mppf) [inline modules/standard/List.chpl:440]: Does this warning need to mention that it can be turned off by setting the config param?

**Review by mppf — APPROVED** [review]
LGTM other than maybe mentioning the config param in the warning

File: modules/standard/List.chpl (hunk 1 above)
Comment (lydia-duncan) [reply modules/standard/List.chpl:440]: Ooooh, good call

**Pull request merged**